.github/workflows/linux_meson.yml

```diff
@@ -248,3 +249,89 @@ jobs:
+    if: "github.repository == 'scipy/scipy' || github.repository == ''"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9]
+
+    #    env:
+    #      NPY_USE_BLAS_ILP64: 1
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Setup Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Ubuntu dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libgmp-dev libmpfr-dev libmpc-dev ccache gfortran
+
+        # Install OpenBLAS a la cibuildwheel.
+        chmod +x tools/wheels/cibw_before_build_linux.sh
+        sudo tools/wheels/cibw_before_build_linux.sh .        
+
+    - name: Caching Python dependencies
+      uses: actions/cache@v3
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-prerelease
+
+    - name: Install Python packages
+      run: |
+        # TODO: unpin cython and move to -pre once issues with cython 3.0
+        #       have been ironed out.
+        python -m pip install ninja meson meson-python wheel click rich_click pydevtool "cython==0.29.33"
+        python -m pip install --pre --upgrade --timeout=60 -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+        python -m pip install --pre --upgrade --timeout=60 pytest pytest-xdist pybind11 mpmath gmpy2 pythran threadpoolctl pooch matplotlib
+
+    - name:  Prepare compiler cache
+      id:    prep-ccache
+      shell: bash
+      run: |
+        mkdir -p "${CCACHE_DIR}"
+        echo "dir=$CCACHE_DIR" >> $GITHUB_OUTPUT
+        NOW=$(date -u +"%F-%T")
+        echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
+
+    - name: Setup compiler cache
+      uses:  actions/cache@v3
+      id:    cache-ccache
+      # Reference: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      # NOTE: The caching strategy is modeled in a way that it will always have a unique cache key for each workflow run
+      # (even if the same workflow is run multiple times). The restore keys are not unique and for a partial match, they will
+      # return the most recently created cache entry, according to the GitHub Action Docs.
+      with:
+        path: ${{ steps.prep-ccache.outputs.dir }}
+        # Restores ccache from either a previous build on this branch or on main
+        key:  ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-prerelease-${{ steps.prep-ccache.outputs.timestamp }}
+        restore-keys: |
+          ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-prerelease-
+
+    - name: Build and install SciPy
+      run: |
+        python dev.py build -j2
+
+    - name: Ccache performance
+      shell: bash -l {0}
+      run: ccache -s
+
+    - name: Test SciPy
+      run: |
+        export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
```

Comment:
We shouldn't ever need this, that makes me think something is wrong. I guess that's because of the custom OpenBLAS location? Can you try `export PKG_CONFIG_PATH=/usr/local/lib` instead?

Reply:
Ah wait, not awake yet - we're testing not building.

Reply:
Self-correcting again, we _are_ building as well as testing. Then `PKG_CONFIG_PATH` should work and is preferred.

Reply:
Ah yes, there's `python dev.py build`. It doesn't quite make sense to me that the build works but import fails at test time - probably something else is wrong.

The way this works is that if a library is in a non-default path, the build should write RPATH entries for those non-default paths into the extension modules so things work at test time. `PKG_CONFIG_PATH` helps only with discovering dependencies at build time. If that works but then afterwards a library path goes missing, that's a bug somewhere. `LD_LIBRARY_PATH` is an anti-pattern that should never be needed. The only reason it's kinda okay (but still not great) in the wheel builds is that afterwards we run `auditwheel`/`delocate` to vendor the relevant shared libraries.

Reply:
We could merge this once CI finishes and clean it up later though, to get on with the Azure migration.

Reply:
CI was green after https://github.com/scipy/scipy/pull/15958/commits/283def421320b2c05ffb6b23b329f0238f46e4a6

The following commits just removed comments.

Reply:
Can we merge @rgommers?